go/mysql/server_flaky_test.go

```diff
@@ -1503,3 +1503,39 @@ func TestServerFlush(t *testing.T) {
 	require.NoError(t, err)
 	assert.Nil(t, row)
 }
+
+func TestTcpKeepAlive(t *testing.T) {
+	th := &testHandler{}
+	l, err := NewListener("tcp", "127.0.0.1:", NewAuthServerNone(), th, 0, 0, false, false, 0)
+	require.NoError(t, err)
+	defer l.Close()
+	go l.Accept()
+
+	host, port := getHostPort(t, l.Addr())
+	params := &ConnParams{
+		Host: host,
+		Port: port,
+	}
+
+	var called bool
+	l.TcpPropFunc = func(conn *net.TCPConn, duration time.Duration) error {
+		called = true
+		return nil
+	}
+
+	// on connect, the tcp method should be called.
+	c, err := Connect(context.Background(), params)
+	require.NoError(t, err)
+	defer c.Close()
+	require.True(t, called, "tcp property method not called")
+
+	// move to original method
+	l.TcpPropFunc = setTcpConnProperties
```

Comment:
Could we do this by storing the original value as a variable and then restoring it in a defer. That would make this part of code resilient to changes in the function later down the line.